File: src/main/java/com/mattiaswikstrom/hyper/Length.java
package com.mattiaswikstrom.hyper;

import java.io.Serializable;

public class Length implements Serializable
{
    final public double l;

    public Length(double l)
    {
        this.l = l;
    }

    public static Length neg(Length l)
    {
        return new Length(-l.l);
    }

    public static Length diff(Length l1, Length l2)
    {
        return new Length(l1.l - l2.l);
    }

    public static Length sum(Length l1, Length l2)
    {
        return new Length(l1.l + l2.l);
    }

    public static Length abs(Length l)
    {
        if (l.l < 0)
            return new Length(-l.l);
        else
            return new Length(l.l);
    }
}

File: hyper-web/src/Length.ts
export class Length {
    readonly l: number;

    constructor(l: number) {
        this.l = l;
    }

    static neg(l: Length): Length {
        return new Length(-l.l);
    }

    static diff(l1: Length, l2: Length): Length {
        return new Length(l1.l - l2.l);
    }

    static sum(l1: Length, l2: Length): Length {
        return new Length(l1.l + l2.l);
    }

    static abs(l: Length): Length {
        if (l.l < 0) {
            return new Length(-l.l);
        } else {
            return new Length(l.l);
        }
    }
}
